packages/lodestar/src/chain/eventHandlers.ts

```diff
@@ -244,32 +236,19 @@ export async function onBlock(
     await this.db.block.add(block);
   }
 
+  // Only process attestations in response to an non-prefinalized block
   if (!job.prefinalized) {
     const attestations = Array.from(readonlyValues(block.message.body.attestations));
 
-    // Only process attestations in response to an non-prefinalized block
-    const indexedAttestations = await Promise.all([
-      // process the attestations in the block
-      ...attestations.map((attestation) => {
-        return this.attestationProcessor.processAttestationJob({
-          attestation,
-          // attestation signatures from blocks have already been verified
-          validSignature: true,
-        });
-      }),
-      // process pending attestations which needed the block
-      ...this.pendingAttestations.getByBlock(blockRoot).map((job) => {
-        this.pendingAttestations.remove(job);
-        return this.attestationProcessor.processAttestationJob(job);
-      }),
-    ]);
-
-    if (this.metrics) {
-      // Register attestations in metrics
-      for (const attestation of indexedAttestations) {
-        if (attestation) {
-          this.metrics.registerAttestationInBlock(attestation, block.message);
-        }
+    for (const attestation of attestations) {
+      try {
+        const indexedAttestation = postState.epochCtx.getIndexedAttestation(attestation);
```

Comment:
we did get indexed attestation to verify signatures. I wonder if we could get it from some caches to avoid creating it again?

Reply:
Right! I though about this optimization too, it should be doable but I would tackle in another PR for sure.

Reply:
agree 👍

Reply:
https://github.com/ChainSafe/lodestar/issues/2771